### PR TITLE
Fix up local tests in fresh environments

### PR DESCRIPTION
**Pull request by bhearsum**
I found that a plain `tox` invocation wasn't working for me. This is in part due to `pin-helper.sh` not doing the right thing for Balrog, but also because `tox` renamed the `whitelist_externals` variable (and didn't maintain backwards compat). This fixed things up for me. Technically, the requirements change could affect prod, but we used to use `--allow-unsafe` for this until https://github.com/mozilla-releng/balrog/commit/6565736cf738c515e5f69c775b738fa1e704c81c (it just wasn't documented...so no wonder it was dropped).

### CODE REVIEW

**Review by JohanLorenzo — APPROVED** [review]
LGTM!

File: requirements/base.in [hunk 1]
```diff
@@ -24,5 +24,6 @@ setuptools
 sqlalchemy
 sqlalchemy-migrate
 statsd
+typing-extensions
```
Comment (jcristau) [inline requirements/base.in:27]: Why is this necessary?

Comment (bhearsum) [reply requirements/base.in:27]: Uhh, whoops. It's not! (I was experimenting with adding types when I noticed this - clearly this got committed to the wrong branch.)

**Review by jcristau — APPROVED** [review]
